android/src/main/java/com/vydia/RNUploader/GlobalRequestObserverDelegate.kt

```diff
@@ -22,16 +24,21 @@ class GlobalRequestObserverDelegate(reactContext: ReactApplicationContext) : Req
   }
 
   override fun onError(context: Context, uploadInfo: UploadInfo, exception: Throwable) {
+    var errorMessage = exception.message
     val params = Arguments.createMap()
     params.putString("id", uploadInfo.uploadId)
 
-    // Make sure we do not try to call getMessage() on a null object
-    if (exception != null) {
-      params.putString("error", exception.message)
-    } else {
-      params.putString("error", "Unknown exception")
+    when (exception) {
+      is UserCancelledUploadException -> {
+        errorMessage = "User cancelled upload"
+      }
+      is UploadError -> {
+        errorMessage = "responseCode=${exception.serverResponse.code} responseBody=${exception.serverResponse.bodyString}"
+        Log.e(TAG, "Error, upload error: ${exception.serverResponse.code} ${exception.serverResponse.bodyString}")
+      }
```

Comment:
This ensures we forward the S3 upload error to the js side, which fixes upload retry if the presigned url is expired.
Will also help know the reason of failure to upload as before we just got a `Upload error`